REGRESSION(263724@main): [ Monterey+ ] TestWebKitAPI.ParserYieldTokenTests.AsyncScriptRunsWhenFetched frequently fails

rniwa opened the pull request:
#### 360db466f1b5b149b60a3050ceb2a5f1cd6906c0
<pre>
REGRESSION(263724@main): [ Monterey+ ] TestWebKitAPI.ParserYieldTokenTests.AsyncScriptRunsWhenFetched frequently fails
<a href="https://bugs.webkit.org/show_bug.cgi?id=256614">https://bugs.webkit.org/show_bug.cgi?id=256614</a>

Reviewed by Wenson Hsieh.

Allow an alternative ordering for the 3rd &amp; 4th messages.

* Tools/TestWebKitAPI/Tests/WebKitCocoa/ParserYieldTokenTests.mm:
(TEST):

Canonical link: <a href="https://commits.webkit.org/263948@main">https://commits.webkit.org/263948@main</a>
</pre>
<!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/dd4ae64a6b692c113d03866eb1ee54b68d17de95

| Misc | iOS, tvOS & watchOS  | macOS  | Linux |  Windows |
| ----- | ---------------------- | ------- |  ----- |  --------- |
| [✅ 🧪 style](https://ews-build.webkit.org/#/builders/38/builds/6186 "Passed style check") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/26/builds/6376 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/14/builds/6557 "Built successfully") | [✅ 🛠 wpe](https://ews-build.webkit.org/#/builders/5/builds/7751 "Built successfully") | [✅ 🛠 wincairo](https://ews-build.webkit.org/#/builders/32/builds/6520 "Built successfully") 
| [✅ 🧪 bindings](https://ews-build.webkit.org/#/builders/9/builds/6186 "Passed tests") | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/23/builds/6596 "Built successfully") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/16/builds/6327 "Built successfully") | [  ~~🧪 wpe-wk2~~](https://ews-build.webkit.org/#/builders/34/builds/9406 "The change is no longer eligible for processing. Commit was outdated when EWS attempted to process it.") | 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/11/builds/6296 "Passed tests") | [✅ 🧪 ios-wk2](https://ews-build.webkit.org/#/builders/15/builds/6317 "Passed tests") | [✅ 🧪 api-mac](https://ews-build.webkit.org/#/builders/18/builds/5603 "Passed tests") | [✅ 🛠 gtk](https://ews-build.webkit.org/#/builders/2/builds/7818 "Built successfully") | 
| | [✅ 🧪 ios-wk2-wpt](https://ews-build.webkit.org/#/builders/39/builds/3791 "Passed tests") | [✅ 🧪 mac-wk1](https://ews-build.webkit.org/#/builders/10/builds/5582 "Passed tests") | [  ~~🧪 gtk-wk2~~](https://ews-build.webkit.org/#/builders/1/builds/13474 "The change is no longer eligible for processing. Commit was outdated when EWS attempted to process it.") | 
| | [✅ 🧪 api-ios](https://ews-build.webkit.org/#/builders/13/builds/5656 "Passed tests") | [✅ 🧪 mac-wk2](https://ews-build.webkit.org/#/builders/36/builds/5662 "Passed tests") | [✅ 🧪 api-gtk](https://ews-build.webkit.org/#/builders/21/builds/7895 "Passed tests") | 
| | [✅ 🛠 tv](https://ews-build.webkit.org/#/builders/7/builds/6132 "Built successfully") | [✅ 🧪 mac-AS-debug-wk2](https://ews-build.webkit.org/#/builders/17/builds/5016 "Passed tests") | | 
| | [✅ 🛠 tv-sim](https://ews-build.webkit.org/#/builders/30/builds/5548 "Built successfully") | | | 
| | [✅ 🛠 watch](https://ews-build.webkit.org/#/builders/4/builds/9693 "Built successfully") | | | 
| [✅ 🛠 🧪 unsafe-merge](https://ews-build.webkit.org/#/builders/22/builds/728 "Built successfully and passed tests") | [✅ 🛠 watch-sim](https://ews-build.webkit.org/#/builders/31/builds/5915 "Built successfully") | | | 
<!--EWS-Status-Bubble-End-->

whsieh commented on Tools/TestWebKitAPI/Tests/WebKitCocoa/ParserYieldTokenTests.mm:
```diff
@@ -192,6 +192,12 @@ static void waitForDelay(Seconds delay)
-    EXPECT_WK_STREQ("Running async script.", eventMessages[2]);
-    EXPECT_WK_STREQ("Finished requesting async script.", eventMessages[3]);
+    if ([eventMessages[2] isEqualToString:@"Running async script."]) {
+        EXPECT_WK_STREQ("Running async script.", eventMessages[2]);
```
Nit - you can omit the `EXPECT_WK_STREQ` check here.

rniwa replied:
Oh, that's a good point. Will do.

rniwa committed:
REGRESSION(263724@main): [ Monterey+ ] TestWebKitAPI.ParserYieldTokenTests.AsyncScriptRunsWhenFetched frequently fails
https://bugs.webkit.org/show_bug.cgi?id=256614

Reviewed by Wenson Hsieh.

Allow an alternative ordering for the 3rd & 4th messages.

* Tools/TestWebKitAPI/Tests/WebKitCocoa/ParserYieldTokenTests.mm:
(TEST):

Canonical link: https://commits.webkit.org/263948@main
